lib/api/goal/ExecuteGoalResult.ts

```
@@ -70,10 +70,32 @@ export interface ExecuteGoalResult extends GoalDetails {
      * 0 is success; non-zero exit codes will mark the goal as failed,
```

Should probably add that 0 or undefined is success.

---

And that you should throw an error on failure.